Assets/Scripts/PlayerScripts/HealthController.cs

```diff
@@ -1,16 +1,25 @@
 using UnityEngine;
 using UnityEngine.UI;
-
-public class Health : MonoBehaviour
+public class HealthController : MonoBehaviour
 {
     [SerializeField] private int _playerHealth = default;
     [SerializeField] private int _healthNumber = default;
     [SerializeField] private Image[] _lifeContainer = default;
     [SerializeField] private Sprite _fullContainer = default;
     [SerializeField] private Sprite _emptyContainer = default;
-
+    [SerializeField] private int _touchDamage = default;
     private void Update()
     {
+        UpdateHealth();
+    }
+
+    public void UpdateHealth()
+    {
+        if (_playerHealth <= 0 )
+        {
+            var playerMovement = gameObject.GetComponent<PlayerMovement>();
```

Comment:
Get this component on Awake and store it to avoid grabbing it every time

Reply:
![image](https://user-images.githubusercontent.com/85502316/156895997-b352c403-8ce6-4765-bcb5-7be30dd56c61.png)
how do i fix this problem, it says that the variable does not exist in the current context